scripts/check_e2e_tests_are_captured_in_ci.py

```diff
@@ -0,0 +1,172 @@
+
+
+def get_e2e_suite_names_from_protractor_file():
+    """Extracts the test suites section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of all test suites in sorted order
+        in protractor.conf.js file.
+    """
+    protractor_config_file_content = read_protractor_conf_file()
+    # The following line extracts suite object from protractor.conf.js.
+    suite_object_string = re.compile(
+        r'suites = {([^}]+)}').findall(protractor_config_file_content)[0]
+
+    # The following line extracts the keys/test suites from the key: value
+    # pair from the suites object.
+    key_regex = re.compile(r'\b([a-zA-Z_-]*):')
+    protractor_suites = key_regex.findall(suite_object_string)
+
+    return sorted(protractor_suites)
+
+
+def read_protractor_conf_file():
+    """Returns the contents of core/tests/protractor.conf.js file.
+
+    Returns:
+        str. The contents of protractor.conf.js, as a string.
+    """
+    protractor_config_file_content = python_utils.open_file(
+        PROTRACTOR_CONF_FILE_PATH, 'r').read()
+    return protractor_config_file_content
+
+
+def read_and_parse_travis_yml_file():
+    """Returns the contents of .travis.yml, as a dict.
+
+    Returns:
+        dict. Contents of the travis.yml file parsed as a dict.
+    """
+    travis_ci_file_content = python_utils.open_file(
+        TRAVIS_CI_FILE_PATH, 'r').read()
+    travis_ci_dict = utils.dict_from_yaml(travis_ci_file_content)
+    return travis_ci_dict
+
+
+def main():
+    """Test the travis ci file and protractor.conf.js to have same
+    e2e test suites.
+    """
+    python_utils.PRINT('Checking e2e tests are captured in travis.yml started')
+    protractor_test_suites = get_e2e_suite_names_from_protractor_file()
+    yaml_jobs = get_e2e_suite_names_from_jobs_travis_yml_file()
+    yaml_scripts = get_e2e_suite_names_from_script_travis_yml_file()
+
+    for excluded_test in TEST_SUITES_NOT_RUN_ON_TRAVIS:
+        protractor_test_suites.remove(excluded_test)
+
+    if not yaml_jobs or not yaml_scripts or not protractor_test_suites:
+        if not yaml_jobs:
+            raise Exception('The e2e test suites that have been extracted from '
+                            'jobs section from travis.ci are empty.')
+        if not yaml_scripts:
+            raise Exception('The e2e test suites that have been extracted from '
+                            'script section from travis.ci are empty.')
+        if not protractor_test_suites:
+            raise Exception('The e2e test suites that have been extracted from '
+                            'protractor.conf.js are empty.')
+
+    if (TRAVIS_PROTRACTOR_COMMON_TEST not in yaml_jobs or
+            TRAVIS_PROTRACTOR_COMMON_TEST not in yaml_scripts or
+            TRAVIS_PROTRACTOR_COMMON_TEST not in protractor_test_suites):
+        if TRAVIS_PROTRACTOR_COMMON_TEST not in yaml_jobs:
+            raise Exception('explorationImprovementsTab is missing from '
```

Comment:
explorationImprovementsTab is expected to be in the e2e test suites extracted from the jobs section of .travis.yml, but it is missing.

(Ditto below, similarly. Please proofread your messages, too: it's .travis.yml not travis.ci. Note the leading period.)

Reply:
Done.